src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,123 @@
 import React from 'react';
 import './TodoList.scss';
+import classNames from 'classnames';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  todos: Todo[],
+  onCheckted: (userId: number) => void,
+  selectedUserId: number,
+};
+
+type State = {
+  titleInput: string,
+  sortBy: string,
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state = {
+    titleInput: '',
+    sortBy: 'all',
+  };
+
+  handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      titleInput: event.target.value.toLowerCase(),
+    });
+  };
+
+  handleSelectedChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({
+      sortBy: event.target.value,
+    });
+  };
+
+  getSortedBySelected = (todos: Todo[]) => {
+    if (this.state.sortBy !== 'all') {
+      if (this.state.sortBy === 'ready') {
+        return todos.filter(todo => todo.completed === true);
+      }
+
+      return todos.filter(todo => todo.completed === false);
+    }
```

Comment:
Чтобы исправить баг №2 нам нужно здесь в фильтре убрать === true/false и написать следующим образом:

```suggestion
      if (this.state.sortBy === 'ready') {
        return todos.filter(todo => todo.completed);
      }

      return todos.filter(todo => !todo.completed);
    }
```
Так мы говорим, что при фильтре по 'ready' нам будут показываться таски, у которых есть поле completed и его значение может быть приведено к тру.
Соответственно, при фильтре по notReady нам будут выдаваться таски, у которых либо нет поля completed, либо его значение может быть приведено к false.

Кроме того, можно было бы эту всю конструкцию переписать на switch case вместо if внутри if. В switch case мы бы сделали 2 кейса, это когда у нас sortBy === ready и sortBy === notReady, а в дефолтном кейсе мы бы отображали всё.